sql3/planner/expressionanalyzer.go

```diff
@@ -151,6 +152,15 @@ func (p *ExecutionPlanner) analyzeExpression(ctx context.Context, expr parser.Ex
 	case *parser.DateLit:
 		return e, nil
 
+	case *parser.SysVariable:
+		// retrieve the system variables store from the context
+		sysvar := ctx.Value(types.SysVarCtxKey).(*types.SysVars)
+		// from the store get the value for the referenced system variable
+		// using a shared store allows us to assign same value for all references of a system variable, especially CURRENT_TIMESTAMP enjoys a consistent value.
+		dt := sysvar.GetSystemVariable(e.Token)
+		e.Value = dt
+		return e, nil
+
```

Comment:
I looked around - it turns out no-one does a single value for 'current_time' variables that's consistent for the request/transaction - you get what you get when you execute it each time. So I don't think this is a requirement.

Reply:
Will work on deferring evaluation to execution and generalizing the variable expression type. 

Evaluating all date time scalar references simultaneously (to same value) within the context of a single statements execution is what ANSI sql specification prescribes (see spec text extracted). I know oracle, postgres adhere to this, not sure about others. Here is an example of the effect:

`insert into demo(id, time_created, last_update) select id, current_timestamp, current_timestamp from ten_million_rows_table;`

All the 10 million rows will have the same timestamp value for time_created and last_update regardless of the time first row is inserted and the last. 

On the other hand, the following block will assign each row a different time:

`declare cursor x for select id from ten_million_rows_table

open cursor x
loop
   fetch x into _id
   insert into demo(id, time_created, last_update) values(_id, current_timestamp, current_timestamp)
end loop
`

Following is the ANSI specification, rule #3 talks about simultaneous evaluation:

        6.8  <datetime value function>

         Function

         Specify a function yielding a value of type datetime.

         <current date value function> ::= CURRENT_DATE

         <current time value function> ::=
                CURRENT_TIME [ <left paren> <time precision> <right paren> ]

         <current timestamp value function> ::=
                CURRENT_TIMESTAMP [ <left paren> <timestamp precision> <right paren> ]

         General Rules

         1) The <datetime value function>s CURRENT_DATE, CURRENT_TIME, and
            CURRENT_TIMESTAMP respectively return the current date, current
            time, and current timestamp; the time and timestamp values are
            returned with time zone displacement equal to the current time
            zone displacement of the SQL-session.

         2) If specified, <time precision> and <timestamp precision> respec-
            tively determine the precision of the time or timestamp value
            returned.

         3) If an SQL-statement generally contains more than one reference
            to one or more <datetime value function>s, then all such ref-
            erences are effectively evaluated simultaneously. The time of
            evaluation of the <datetime value function> during the execution
            of the SQL-statement is implementation-dependent.

Reply:
Not saying we shouldn't do simultaneous eval - saying it's going to be complex to get right in a distributed env, and asking if we really need to take the time to do it now. I'd argue that we can defer and move on - we have a bunch of other stuff that is probably more impactful in our ability to be successful.